From 70c902628f7d9e757b88445b1bfb3841e88ff2f9 Mon Sep 17 00:00:00 2001
From: Sergey V. Udaltsov <svu@gnome.org>
Date: Mon, 11 Jan 2010 11:20:13 +0000
Subject: Extra parameter for xkl_engine_stop_listen

New version of libxklavier (5.0) requires to specify flags when you stop
listening
---
diff --git a/plugins/keyboard/gsd-keyboard-xkb.c b/plugins/keyboard/gsd-keyboard-xkb.c
index a86fc80..a7fb5b5 100644
--- a/plugins/keyboard/gsd-keyboard-xkb.c
+++ b/plugins/keyboard/gsd-keyboard-xkb.c
@@ -892,7 +892,9 @@ gsd_keyboard_xkb_shutdown (void)
 	if (!inited_ok)
 		return;
 
-	xkl_engine_stop_listen (xkl_engine);
+	xkl_engine_stop_listen (xkl_engine,
+				XKLL_MANAGE_LAYOUTS |
+				XKLL_MANAGE_WINDOW_STATES);
 
 	gdk_window_remove_filter (NULL, (GdkFilterFunc)
 				  gsd_keyboard_xkb_evt_filter, NULL);
--
cgit v0.8.3.1
